### ISSUE TITLE
Select all only selects the viewable text before scrolling

### ISSUE THREAD

**Issue by LiamDawe** [mate-desktop/mate-terminal#372]
#### Expected behaviour

Select all should actually select ALL text, not just what's on the screen before the cut-off.

#### Actual behaviour

It only selects what's currently seen.

#### Steps to reproduce the behaviour

Have something really long in terminal and then try to select all, see what happens?

#### MATE general version

1.24.1-1

#### Package version

1.24.1-1

#### Linux Distribution

EndeavorOS (Arch), latest updates

#### Link to bugreport of your Distribution (requirement)

I don't know if it's a distribution issue or intentional, won't make one until people working on the actual app say what is intentional otherwise it's a waste of everyone's time.

**Comment by lukefromdc** [comment]
I have NOT had this problem in mate-terminal from git master, now or when master was on 1.24. Wonder what is different?

**Comment by flexiondotorg** [comment]
Just tested this on Ubuntu MATE 20.04.1 using mate-terminal 1.24.1 and can not reproduce the issue.

**Comment by rbuj** [comment]
culprit https://gitlab.gnome.org/GNOME/vte/-/commit/caf8a6a7a17a3f70fa5fc611c71ffb27a5cfee75

**Comment by JoakimCh** [comment]
@rbuj What could/will be done about it?

I also have this issue after updating my Debian Bullseye install.
Mate Terminal here is also version 1.24.1-1 and MATE 1.24.1.

Very annoying since I often have to copy all the output and store it for later.

**Comment by rbuj** [comment]
> @rbuj What could/will be done about it?

You can file a bug report at https://gitlab.gnome.org/GNOME/vte/-/issues

**Comment by N0rbert** [comment]
Ubuntu MATE 21.04 is affected, see https://pad.lv/1922276 . I'm shocked.

**Comment by lukefromdc** [comment]
I have this problem now, strongly suspect a VTE update caused this, will try rolling back libvte to check

**Comment by lukefromdc** [comment]
I just CONFIRMED rolling back libvte to Debian's
libvte-2.91-0_0.62.1-1_amd64.deb
fixes this and restores the ability to cut and paste text that has been scrolled past

**Comment by N0rbert** [comment]
Reverting VTE on Ubuntu 21.04 to version *0.62.0-1ubuntu1* from Ubuntu 20.10 helps.

**Comment by N0rbert** [comment]
Ubuntu MATE 21.10 impish will be affected by this bug if no additional measures will be taken. Please do something :cry:

**Comment by lukefromdc** [comment]
I suspect this is another VTE issue. I have kept an older version of VTE on my system since the version of it that broke scrolling support entirely.Are you able to scroll back at all? If so VTE has been partially but not entirely fixed.

Both mate-terminal and gnome-terminal (from which mate-terminal was forked) are basically wrappers around VTE,  which is not our package.

**Comment by JoakimCh** [comment]
> I suspect this is another VTE issue. I have kept an older version of VTE on my system since the version of it that broke scrolling support entirely.Are you able to scroll back at all? If so VTE has been partially but not entirely fixed. Both mate-terminal and gnome-terminal (from which mate-terminal was forked) are basically wrappers around VTE, which is not our package.

Yeah, someone already linked to the commit which destroyed it:
> Including the scrollback in select-all makes it too easy to select so much
data that putting it on the clipboard either hangs the process for a long time
or causes a crash:
https://gitlab.gnome.org/GNOME/vte/-/commit/caf8a6a7a17a3f70fa5fc611c71ffb27a5cfee75

I don't know why anyone would accept such a stupid solution requiring another fix, yes he put "FIXME!" in a comment...

**Comment by allanlaal** [comment]
main issue for tracking: https://gitlab.gnome.org/GNOME/vte/-/issues/2504

**Comment by N0rbert** [comment]
We should do something with these smart evil geniuses from GNOME. This bug should not appear in Ubuntu MATE 22.04 LTS.

**Comment by N0rbert** [comment]
Ubuntu MATE 22.04 LTS is still affected.

**Comment by DennisPeter** [comment]
Yes Ubuntu Mate 22.04 LTS is affected as of latest on May 17, 2022.

It's silly how this regression got in there in the first place.
A workaround can be to redirect the output to a file: (example):

make &> buildlog.txt

Or another workaround:
Scroll to top of terminal.  Highlight some letters with your mouse.  Scroll to bottom.  Hold down shift and click bottom of text in terminal.  All should be highlited now.
Now you can right-click and choose Copy. (or CTRL+SHIFT+C)

This regression is bad as it breaks our automated scripts to grab data off the terminal.  Now have a couple of devs looking for alternatives (terminals or re-writing the scripts to get the data).  Ugh...

**Comment by N0rbert** [comment]
Better temporary fix for Ubuntu MATE 22.04 LTS is reverting VTE to previous non-buggy version:

> To fix the 1922276 bug one can use 20.04 LTS's VTE library from [my PPA](https://launchpad.net/~nrbrtx/+archive/ubuntu/vte?field.series_filter=jammy) as follows:
> ```
> sudo add-apt-repository ppa:nrbrtx/vte
> sudo apt-get update
> sudo apt-get dist-upgrade
>````

(The method shown above was originally posted at https://ubuntu-mate.community/t/howto-fixing-some-ubuntu-mate-22-04-lts-issues/25345 ) .

**Comment by DennisPeter** [comment]
Hi N0rbert,
Thank you!  Your VTE PPA is working great.
Ah... I was unaware of the forum.  I'm a software dev myself so naturally I head to the source code.  Thanks for the link!
You would think terminal emulation is a mature technology that only needs the odd security update, but I guess not.

**Comment by N0rbert** [comment]
Both Debian 11 and 12 are still affected by this bug. Reported at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1033495 .
